packages/mobile/src/fiatExchanges/utils.tsx

```diff
@@ -80,3 +119,33 @@ export const openTransak = (
     currencyToBuy,
   })
 }
+
+type ProviderAvailability = typeof providerAvailability
+type SpecificProviderAvailability = { [K in keyof ProviderAvailability]: boolean }
+
+type Entries<T> = Array<{ [K in keyof T]: [K, T[K]] }[keyof T]>
+
+export function getProviderAvailability(
+  userLocation: UserLocation | undefined
+): SpecificProviderAvailability {
+  const { countryCodeAlpha2, stateCode } = userLocation
+    ? { countryCodeAlpha2: userLocation.country, stateCode: userLocation.state }
+    : { countryCodeAlpha2: null, stateCode: null }
+
+  // tslint:disable-next-line: no-object-literal-type-assertion
+  const features = {} as SpecificProviderAvailability
+  for (const [key, value] of Object.entries(providerAvailability) as Entries<
+    ProviderAvailability
+  >) {
+    if (!countryCodeAlpha2) {
+      features[key] = false
+    } else {
+      if (countryCodeAlpha2 === 'US' && (value as any)[countryCodeAlpha2] !== true) {
```

Comment:
I find all the `value as any` casts a bit annoying, not sure if there's a good way to avoid all of them, maybe just do it once and assign to a variable with a more descriptive name than `value`?

Reply:
I messed around with the types a little bit to eliminate the casting but couldn't figure it out. I don't really want to spend much more time on this given it's mostly stylistic (though it bothers me too) but if you have a solution I am definitely willing to incorporate it